inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -745,18 +771,400 @@ class CompareNodesAttributes {
         }
         if (!c.m_compare_attr.all_attr_was_compared()) {
             return "not all of attr was compared: " +
-                   std::to_string(c.m_compare_attr.compared_attr_number()) + " vs " +
-                   std::to_string(c.m_store_attr.attributes_number());
+                   to_str(c.m_compare_attr.compared_attr_number()) + " vs " +
+                   to_str(c.m_store_attr.attributes_number());
         }
-        return "looks good [compared " + std::to_string(c.m_compare_attr.compared_attr_number()) +
+        return "looks good [compared " + to_str(c.m_compare_attr.compared_attr_number()) +
                " attributes]";
     }
 
 private:
-    attr_comparison::ReadAndStoreAttributes m_store_attr;
-    attr_comparison::ReadAndCompareAttributes m_compare_attr;
+    ReadAndStoreAttributes m_store_attr;
+    ReadAndCompareAttributes m_compare_attr;
 };
 
+}  // namespace detail
+
+Comparator::Result compare(
+    ngraph::Node* node1, ngraph::Node* node2, Comparator::CmpValues comparition_flags) {
+    detail::CompareNodesAttributes compare_nodes_attr(comparition_flags);
+    node1->visit_attributes(compare_nodes_attr.get_ref_reader());
+    node2->visit_attributes(compare_nodes_attr.get_cmp_reader());
+    if (!compare_nodes_attr.equal()) {
+        return Comparator::Result::error(
+            "Comparison of attributes failed for nodes " + name(node1) + ", " + name(node2) +
+            " [cmp status: " + to_str(compare_nodes_attr) + "]");
+    }
+    return Comparator::Result::ok(to_str(compare_nodes_attr));
+}
+
+}  // namespace attributes
+
+namespace subgraph {
+
+namespace detail {
+
+template <typename Ptr>
+Ptr not_null(Ptr&& p) {
+    if (!p) {
+        throw std::runtime_error{"empty ptr"};
```

Comment:
I think in the IE tests you can throw Inference Engine exceptions and use the macro `THROW_IE_EXCEPTION << "empty ptr";`

Reply:
OK